data-prepper-plugins/grok-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/grok/GrokProcessor.java

```diff
@@ -258,6 +260,9 @@ private void matchAndMerge(final Event event) {
         }
 
         if (grokkedCaptures.isEmpty()) {
+            if (matchFailTag != null) {
```

Comment:
If this parameter is a `Set`, it could default to be an empty set. and we could just call an `addTags` method on Event to add all tags regardless

Reply:
Please see above. I do not see the need for making it a set.